src/transform/FooterMenu.js

```diff
@@ -0,0 +1,215 @@
+import './FooterMenu.css'
+
+/**
+ * @typedef {function} FooterMenuItemPayloadExtractor
+ * @param {!Document} document
+ * @return {!string[]} Important - should return empty array if no payload strings.
+ */
+
+/**
+  * @typedef {function} FooterMenuItemClickCallback
+  * @param  {!string[]} payload Important - should return empty array if no payload strings.
+  * @return {void}
+  */
+
+/**
+ * Extracts array of no-html page issues strings from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const pageIssuesStringsArray = document => {
+  const tables = document.querySelectorAll(
+    'div#content_block_0 table.ambox:not(.ambox-multiple_issues):not(.ambox-notice)'
+  )
+  // Get the tables into a fragment so we can remove some elements without triggering a layout
+  const fragment = document.createDocumentFragment()
+  for (let i = 0; i < tables.length; i++) {
+    fragment.appendChild(tables[i].cloneNode(true))
+  }
+  // Remove some element so their text doesn't appear when we use "innerText"
+  Array.from(fragment.querySelectorAll('.hide-when-compact, .collapsed')).forEach(el => el.remove())
+  // Get the innerText
+  return Array.from(fragment.querySelectorAll('td[class$=mbox-text]')).map(el => el.innerText)
+}
+
+/**
+ * Extracts array of disambiguation page urls from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const disambiguationTitlesArray = document => Array.from(
+  document.querySelectorAll(
+    'div#content_block_0 div.hatnote a[href]:not([href=""]):not([redlink="1"])'
+  )
+).map(el => el.href)
+
+/**
+ * Type representing kinds of menu items.
+ * @type {Object}
+ */
+const MenuItemType = {
+  languages: 1,
+  lastEdited: 2,
+  pageIssues: 3,
+  disambiguation: 4,
+  coordinate: 5
+}
+
+/**
+ * Menu item model.
+ */
+class WMFMenuItem {
+  /**
+   * WMFMenuItem constructor.
+   * @param  {!string} title
+   * @param  {?string} subtitle
+   * @param  {!MenuItemType} itemType
+   * @param  {FooterMenuItemClickCallback} clickHandler
+   * @return {void}
+   */
+  constructor(title, subtitle, itemType, clickHandler) {
+    this.title = title
+    this.subtitle = subtitle
+    this.itemType = itemType
+    this.clickHandler = clickHandler
+    this.payload = []
+  }
+
+  /**
+   * Returns icon CSS class for this menu item based on its type.
+   * @return {!string}
+   */
+  iconClass() {
+    switch (this.itemType) {
```

Comment:
The reason is the elimination of conditional logic and the potential to fall out of sync between the conditional and the declaration, adding new entries is one line instead of a couple, and everything that embodies a menu item's distinction from other items is in one place.

WRT the discussion on how payloadExtractor() would look after this change, you just tack on the ID getter on to the switch and cases (if it's not possible to also embed this conditional into MenuItemType itself):

```js
  payloadExtractor() {
    switch (this.itemType.id) {
      case MenuItemType.pageIssues.id: return pageIssuesStringsArray
      case MenuItemType.disambiguation.id: return disambiguationTitlesArray
      default: return undefined
    }
  }
```